File: tests/integration/benchmarks/fixtures/rust/rosetta/src/pipeline_test.rs
//! Integration tests for the complete data processing pipeline.
//!
//! This test file exercises NON-name-matched test detection:
//! pipeline_test.rs imports processor and models, so those files
//! should appear as "test-backed" in the Explorer even without
//! a directly name-matched test file.

#[cfg(test)]
mod tests {
    use crate::processor::{run, summarize};
    use crate::models::{Record, validate_config};
    use crate::types::ProcessorConfig;

    #[test]
    fn processes_records_through_complete_pipeline() {
        // Create test records using models factory
        let records = vec![
            Record::create(1, "Alpha".to_string(), 100.0),
            Record::create(2, "Beta".to_string(), 200.0),
            Record::create(3, "Gamma".to_string(), 300.0),
        ];

        // Process through processor
        let report = run(&records, None);

        // Verify report structure
        assert!((report.total - 600.0).abs() < 0.001);
        assert!((report.average - 200.0).abs() < 0.001);
        assert_eq!(report.records.len(), 3);

        // Verify summarization
        let summary = summarize(&report);
        assert!(summary.contains("600"));
    }

    #[test]
    fn validates_configuration_before_processing() {
        let valid_config = ProcessorConfig {
            batch_size: 100,
            timeout: 5000,
            strict: true,
        };

        let invalid_config = ProcessorConfig {
            batch_size: -1,
            timeout: 0,
            strict: false,
        };

        assert!(validate_config(&valid_config));
        assert!(!validate_config(&invalid_config));
    }

    #[test]
    fn handles_edge_cases_in_pipeline() {
        // Empty input
        let empty_report = run(&vec![], None);
        assert!((empty_report.total - 0.0).abs() < 0.001);

        // Single record
        let single_record = vec![Record::create(1, "Solo".to_string(), 42.0)];
        let single_report = run(&single_record, None);
        assert!((single_report.average - 42.0).abs() < 0.001);
    }
}

File: tests/integration/benchmarks/fixtures/java/rosetta/src/com/rosetta/app/PipelineTest.java
package com.rosetta.app;

import com.rosetta.models.ModelFactory;
import com.rosetta.models.Record;
import com.rosetta.models.Report;
import com.rosetta.processor.Processor;
import com.rosetta.types.ProcessorConfig;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Nested;
import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

/**
 * Integration tests for the complete data processing pipeline.
 *
 * This test file exercises NON-name-matched test detection:
 * PipelineTest.java imports Processor and Record/Report, so those files
 * should appear as "test-backed" in the Explorer even without
 * a directly name-matched test file.
 */
public class PipelineTest {

    @Nested
    @DisplayName("Pipeline Integration")
    class PipelineIntegration {

        @Test
        @DisplayName("processes records through complete pipeline")
        void processesRecordsThroughCompletePipeline() {
            // Create test records using models factory
            List<Record> records = Arrays.asList(
                    Record.create(1, "Alpha", 100.0),
                    Record.create(2, "Beta", 200.0),
                    Record.create(3, "Gamma", 300.0));

            // Process through processor
            Report report = Processor.run(records, null);

            // Verify report structure
            assertEquals(600.0, report.getTotal(), 0.001);
            assertEquals(200.0, report.getAverage(), 0.001);
            assertEquals(3, report.getRecords().size());

            // Verify summarization
            String summary = Processor.summarize(report);
            assertTrue(summary.contains("600"));
        }

        @Test
        @DisplayName("validates configuration before processing")
        void validatesConfigurationBeforeProcessing() {
            ProcessorConfig validConfig = new ProcessorConfig(100, 5000, true);
            ProcessorConfig invalidConfig = new ProcessorConfig(-1, 0, false);

            assertTrue(ModelFactory.validateConfig(validConfig));
            assertFalse(ModelFactory.validateConfig(invalidConfig));
        }

        @Test
        @DisplayName("handles edge cases in pipeline")
        void handlesEdgeCasesInPipeline() {
            // Empty input
            Report emptyReport = Processor.run(Collections.emptyList(), null);
            assertEquals(0.0, emptyReport.getTotal(), 0.001);

            // Single record
            List<Record> singleRecord = Arrays.asList(Record.create(1, "Solo", 42.0));
            Report singleReport = Processor.run(singleRecord, null);
            assertEquals(42.0, singleReport.getAverage(), 0.001);
        }
    }
}
